File: DevSupportAndroid/src/main/java/com/sup/dev/android/views/watchers/TextWatcherRemoveHTML.java
package com.sup.dev.android.views.watchers;

import android.text.SpannableStringBuilder;

import com.sup.dev.java.libs.debug.Debug;

public class TextWatcherRemoveHTML extends BaseTextWatcher {

    private boolean ignoreNext;

    @Override
    public void onTextChanged(CharSequence charSequence, int start, int before, int count) {
        if (!ignoreNext
                && count - before > 1 // Чтоб не тригеррилось при обычном вводе
                && charSequence instanceof SpannableStringBuilder) {
            String text = charSequence.toString();
            SpannableStringBuilder s = (SpannableStringBuilder) charSequence;
            s.clear();
            ignoreNext = true;
            s.append(text);
            ignoreNext = false;
        }
    }
}

File: DevSupportAndroid/src/main/java/com/sup/dev/android/views/watchers/TextWatcherRemoveHTML.kt
package com.sup.dev.android.views.watchers

import android.text.SpannableStringBuilder

class TextWatcherRemoveHTML : BaseTextWatcher() {

    private var ignoreNext: Boolean = false

    override fun onTextChanged(charSequence: CharSequence, start: Int, before: Int, count: Int) {
        if (!ignoreNext
                && count - before > 1 // Чтоб не тригеррилось при обычном вводе
                && charSequence is SpannableStringBuilder) {
            val text = charSequence.toString()
            charSequence.clear()
            ignoreNext = true
            charSequence.append(text)
            ignoreNext = false
        }
    }
}
